src/sentry/ingest/consumer/factory.py

```diff
@@ -159,3 +159,58 @@ def shutdown(self) -> None:
         self._pool.close()
         if self._attachments_pool:
             self._attachments_pool.close()
+
+
+class IngestTransactionsStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    """
+    Processes transactions in either celery or no-celery mode.
+    Transactions are either dispatched to `save_transaction_event` or stored directly in the
+    consumer depending on the mode.
+    """
+
+    def __init__(
+        self,
+        reprocess_only_stuck_events: bool,
+        stop_at_timestamp: int | None,
+        num_processes: int,
+        max_batch_size: int,
+        max_batch_time: int,
+        input_block_size: int | None,
+        output_block_size: int | None,
+        no_celery_mode: bool = False,
+    ):
+        self.consumer_type = ConsumerType.Transactions
```

Comment:
If this is a constant, please use the enum value at line 208. Even if this is used for a test, this is code that will run in production during the test.

Reply:
sorry i'm not sure i follow -- line 208 contains a reference to this `self.consumer_type` variable already